src/rpcblockchain.cpp

```diff
@@ -632,6 +633,87 @@ Value getchaintips(const Array& params, bool fHelp)
     return res;
 }
 
+Value getfeeinfo(const Array& params, bool fHelp)
+{
+    if (fHelp || params.size() != 1)
+        throw runtime_error(
+                "getfeeinfo\n"
+                        "\nReturns details of transaction fees over the last n blocks.\n"
+                        "\nArguments:\n"
+                        "1. blocks     (int, required) the number of blocks to get transaction data from\n"
+                        "\nResult:\n"
+                        "{\n"
+                        "  \"txcount\": xxxxx                (numeric) Current tx count\n"
+                        "  \"txbytes\": xxxxx                (numeric) Sum of all tx sizes\n"
+                        "  \"ttlfee\": xxxxx                 (numeric) Sum of all fees\n"
+                        "  \"feeperkb\": xxxxx               (numeric) Average fee per kb over the block range\n"
+                        "  \"rec_highpriorityfee_perkb\": xxxxx    (numeric) Recommended fee per kb to use for a high priority tx\n"
+                        "}\n"
+                        "\nExamples:\n" +
+                HelpExampleCli("getfeeinfo", "5") + HelpExampleRpc("getfeeinfo", "5"));
+
+
+    int nBlocks = params[0].get_int();
+    int nStartHeight = chainActive.Height() - nBlocks;
+    if (nStartHeight <= 0)
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "invalid start height");
+
+    int nBestHeight = chainActive.Height();
+    CAmount nFees = 0;
+    int64_t nBytes = 0;
+    int64_t nTotal = 0;
+    for (int i = nStartHeight; i <= nBestHeight; i++) {
+        CBlockIndex* pindex = chainActive[i];
+        CBlock block;
+        if (!ReadBlockFromDisk(block, pindex))
+            throw JSONRPCError(RPC_DATABASE_ERROR, "failed to read block from disk");
+
+        CAmount nValueIn = 0;
+        CAmount nValueOut = 0;
+        for (const CTransaction& tx : block.vtx) {
+            if (tx.IsCoinBase() || tx.IsCoinStake())
```

Comment:
Is it intentional to include the fees paid for ZerocoinMint type transactions? (also SwiftX type?) I suppose this goes back to the SmartFee feature not being able to differentiate between "normal" and static-fee based transaction types.

If the intention here is to simply aggregate all fees paid in the block range, then it is correct as-is. However, if the intention is to provide an estimated fee rate for *future* transactions then it is certainly skewed by ZerocoinMint and SwiftX type transactions...but that may be a different issue entirely.

Reply:
The goal is to provide a high priority fee rate to services that need transaction priority.